daffodil-codegen-c/src/main/scala/org/apache/daffodil/codegen/c/generators/CodeGeneratorState.scala

```diff
@@ -354,19 +365,20 @@ class CodeGeneratorState(private val root: ElementBase) {
          |// Initialize, parse, and unparse nodes of the infoset
          |
          |$finalImplementation
-         |// Return a root element for parsing or unparsing the infoset
+         |// Get an infoset (optionally clearing it first) for parsing/walking
          |
          |InfosetBase *
-         |rootElement(void)
+         |get_infoset(bool clear_infoset)
          |{
-         |    static bool initialized;
-         |    static $rootName root;
-         |    if (!initialized)
+         |    static $rootName infoset;
+         |
+         |    if (clear_infoset)
          |    {
-         |        ${rootName}_initERD(&root, (InfosetBase *)&root);
-         |        initialized = true;
+         |        memset(&infoset, 0, sizeof(infoset));
```

Comment:
Is it necessary to walk through the infoset and free any pointers that point to heap allocated memory to avoid memory leaks?

It looks like maybe that's only hexBinary? That's the only place I see malloc. Are only bounded arrays supported by runtime1 an so don't need heap allocations?

Reply:
I added a comment at this location to warn callers that get_infoset(CLEAR_INFOSET) won't free malloc'ed storage for hexBinary prefixed length elements for now.  Writing and calling such extra code didn't seem worth it to me at this time.